### DISCUSSION TITLE
RTSP on Pi Zero 2 W

### DISCUSSION THREAD

**Discussion by beavykin** [bluenviron/mediamtx#4061] [Questions]
### Question

Hi,

I’ve been trying to get my camera to work with audio for a while now and with no luck.

I’m able to view the video with `source: rpiCamera` but only over webRTC, if I try and view it with rtsp or anything else it doesn’t work.

I do want to use webRTC but the code for adding the audio seems to rely on rtsp so it isn’t working.

Am I missing a step to get rtsp working? I’ve tried gstreamer and ffmpeg amongst other things.

Any help would be greatly appreciated!

**Comment by aler9** [comment]
Hello, instructions for adding audio present in the README rely on GStreamer and use the RTSP protocol for pulling video, enhancing it with audio and sending everything back to the server, but they do not bound you to the RTSP protocol, since in MediaMTX conversion between protocols is automatic, so you can publish with RTSP and read with WebRTC without any additional configuration required.

Therefore, this is not the problem.

Unfortunately, nothing else can be asserted since you did not include configuration, server logs, software version or any other thing that might allow to replicate the issue. If you still have issues, open another question and provide a way to replicate the problem.
